### PR TITLE
Add bidirectional checkbox and reverse-card fields to Edit Word dialog

### PR DESCRIPTION
**Pull request by Vladyslav-Soldatenko**
## Summary
- `EditWordDialog` now has the same "bidirectional" checkbox + optional "Customize reverse card" fields as the Add Word screen, reusing the existing `BidirectionalOptionSection` composable rather than duplicating it.
- Extracted the shared checkbox toggle/clear-on-uncheck logic and the blank-to-null override mapping into a new `BidirectionalFields` type (`ui/BidirectionalFields.kt`), and refactored `AddWordUiState`'s equivalent functions to delegate to it — same behavior, single source of truth.
- When editing a word that already has non-blank reverse overrides, the customize section now starts expanded so the user sees what's already customized without an extra click.
- **Repository fix**: enabling bidirectional via edit on a word that's already been reviewed forward previously had no way to ever get its backward card scheduled (the natural "first-ever review" seed in `reviewVocabularyItem` can't fire again for such a row). `VocabularyRepositoryImpl.updateVocabularyItem` now seeds `backwardFsrsDueAt` directly in that specific case, without touching already-new rows or already-scheduled backward progress.
- Localized `EditWordDialog`'s Save/Cancel buttons to string resources (`action_save`, `action_cancel`), matching every other dialog in the file; added translations for `action_save` across all supported locales.

Two gaps were investigated and deliberately left out of this change (documented in the plan, not silently dropped): renaming a word via edit to collide with another word's text can throw an uncaught unique-constraint error (pre-existing, unrelated to bidirectional, needs its own duplicate-detection UX like Add-word's).

## Test plan
- [x] New `BidirectionalFieldsTest.kt` — pure unit tests for the shared toggle/clear/`toCardOptions` logic, including whitespace-only overrides and the "uncheck then re-check doesn't restore cleared text" edge case.
- [x] Extended `WordListScreenTest.kt` — Compose UI tests covering checkbox state, customize-section expand/collapse defaults, save/cancel behavior with and without overrides, and whitespace-only override handling.
- [x] Extended `VocabularyRepositoryBidirectionalTest.kt` — repository tests for the new backward-seed-on-edit behavior, confirming it doesn't fire for brand-new rows, doesn't clobber already-scheduled backward progress, and doesn't affect disabling bidirectional.
- [x] `./gradlew check` (ktlint, detekt, lintDebug, unit tests) passes clean.

---
_Generated by [Claude Code](https://claude.ai/code/session_01PH7k3TXaXXhmKcEd1bs89q)_

### CODE REVIEW

File: app/src/test/java/com/procrastilearn/app/data/repository/VocabularyRepositoryBidirectionalTest.kt [hunk 1]
```diff
@@ -595,4 +595,130 @@ class VocabularyRepositoryBidirectionalTest {
...
+            assertThat(entity.backwardFsrsDueAt).isEqualTo(scheduledBackwardDueAt)
+        }
+
+    @Test
+    fun `updateVocabularyItem persists and clears the reverse override fields`() =
+        runTest {
+            val id = insertVocabulary("run", bidirectional = true)
+
+            repository.updateVocabularyItem(
+                com.procrastilearn.app.domain.model.VocabularyItem(
```
Comment (Vladyslav-Soldatenko) [inline app/src/test/java/com/procrastilearn/app/data/repository/VocabularyRepositoryBidirectionalTest.kt]: here and around it - import, don't use full name
